File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.js
var bestClosingTime = function(customers) {
    let maxi = 0;
    let balance = 0;
    let answer = 0
    for(var i = 0;i < customers.length;i++){
        if(customers[i] === "Y"){
            balance++
        }
        else balance--
        if(balance > maxi){
            maxi = balance;
            answer = i + 1
        }
    }
    return answer;
};

File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        ans = 0 
        prefix = least = customers.count('Y')
        for i, ch in enumerate(customers): 
            if ch == 'N': prefix += 1
            else: prefix -= 1; 
            if prefix < least: 
                ans = i+1
                least = prefix 
        return ans 